plugins/modules/cloudtrail_info.py

```diff
@@ -0,0 +1,245 @@
+            type: bool
+            sample: true
+        latest_delivery_error:
+            description: Displays any Amazon S3 error that CloudTrail encountered when attempting to deliver log files to the designated bucket.
+            type: str
+        latest_notification_error:
+            description: Displays any Amazon SNS error that CloudTrail encountered when attempting to send a notification.
+            type: str
+        latest_delivery_time:
+            description: Specifies the date and time that CloudTrail last delivered log files to an account's Amazon S3 bucket.
+            type: str
+        start_logging_time:
+            description: Specifies the most recent date and time when CloudTrail started recording API calls for an AWS account.
+            type: str
+        stop_logging_time:
+            description: Specifies the most recent date and time when CloudTrail stopped recording API calls for an AWS account.
+            type: str
+        latest_cloud_watch_logs_delivery_error:
+            description: Displays any CloudWatch Logs error that CloudTrail encountered when attempting to deliver logs to CloudWatch Logs.
+            type: str
+        latest_cloud_watch_logs_delivery_time:
+            description: Displays the most recent date and time when CloudTrail delivered logs to CloudWatch Logs.
+            type: str
+        latest_digest_delivery_time:
+            description: Specifies the date and time that CloudTrail last delivered a digest file to an account's Amazon S3 bucket.
+            type: str
+        latest_digest_delivery_error:
+            description: Displays any Amazon S3 error that CloudTrail encountered when attempting to deliver a digest file to the designated bucket.
+            type: str
+        resource_id:
+            description: Specifies the ARN of the resource.
+            type: str
+        tags:
+            description: Any tags assigned to the cloudtrail.
+            type: dict
+            returned: always
+            sample: "{ 'my_tag_key': 'my_tag_value' }"
+
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_filter_list
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+
+
+def get_trails(connection, module):
+    all_trails = []
+    next = False
+    try:
+        result = connection.list_trails()
+    except botocore.exceptions.ClientError as e:
+        module.fail_json_aws(e, msg="Failed to get the cloudtrails.")
+    if result:
+        while not all_trails or result.get("NextToken", ""):
```

Comment:
You can use a paginator object here instead: https://github.com/ansible-collections/amazon.aws/blob/main/docs/docsite/rst/dev_guidelines.rst#api-throttling-rate-limiting-and-pagination

```python
import boto3
cloudtrail = boto3.client('cloudtrail')
paginator = cloudtrail.get_paginator('list_trails')
for trail in paginator.paginate():
    print(trail)
```